copylot/hardware/cameras/flir/flir_camera.py

```diff
@@ -0,0 +1,484 @@
+        """
+        Get the image size nodes for the current camera
+        """
+        self.initialize()
+        nodemap = self.cam.GetNodeMap()
+        node_width = PySpin.CIntegerPtr(nodemap.GetNode('Width'))
+        node_height = PySpin.CIntegerPtr(nodemap.GetNode('Height'))
+        if not PySpin.IsReadable(node_width) and PySpin.IsWritable(node_width):
+            logger.error('Width node is not accessible')
+        if not PySpin.IsReadable(node_height) and PySpin.IsWritable(node_height):
+            logger.error('Height node is not accessible')
+        return node_width, node_height
+
+    @property
+    def image_size(self):
+        """
+        Return the (width, height) of the most recent image size setting in pixels (type: int)
+        """
+        node_width, node_height = self.image_nodes()
+        return node_width.GetValue(), node_height.GetValue()
+
+    @image_size.setter
+    def image_size(self, size):
+        """
+        Set the image size of the current camera
+
+        Parameters
+        ----------
+        size: tuple for image dimensions in pixels (width, height). Type: int
+        """
+        node_width, node_height = self.image_nodes()
+
+        # ensure input is within bounds
+        minw, maxw, minh, maxh = self.image_size_limits
+        if minw <= size[0] <= maxw and minh <= size[1] <= maxh:
+            node_width.SetValue(size[0])
+            node_height.SetValue(size[1])
+        else:
+            logger.error('Input image size is out of bounds. Cannot change settings.')
+
+    @property
+    def image_size_limits(self):
+        """
+        Return the image size limits. Type: int
+        """
+        node_width, node_height = self.image_nodes()
+        return (
+            node_width.GetMin(),
+            node_width.GetMax(),
+            node_height.GetMin(),
+            node_height.GetMax(),
+        )
+
+    @property
+    def binning(self):
+        self.initialize()
+        return (
+            self.cam.BinningHorizontal.GetValue(),
+            self.cam.BinningVertical.GetValue(),
+        )
+
+    @binning.setter
+    def binning(self, val):
+        """
+        Assumes (x,y) binning input
+        """
+        # CHECK FOR ACCESS ERRORS
+        xmin = self.cam.BinningHorizontal.GetMin()
+        xmax = self.cam.BinningHorizontal.GetMax()
+        ymin = self.cam.BinningVertical.GetMin()
+        ymax = self.cam.BinningVertical.GetMax()
+
+        if xmin <= val[0] <= xmax and ymin <= val[1] <= ymax:
+            self.cam.BinningHorizontal.SetValue(val[0])
+            self.cam.BinningVertical.SetValue(val[1])
+        else:
+            logger.error('Input binning is out of bounds. Cannot change settings.')
+
+    @property
+    def shutter_mode(self):
+        pass
+
+    @shutter_mode.setter
+    def shutter_mode(self, mode):
+        pass
+
+    @property
+    def bitdepth(self):
+        pass
+
+    @bitdepth.setter
+    def bitdepth(self, bit):
+        pass
```

Comment:
Could you add a trigger configuration function ? and change the way camera snap an image according to the trigger mode that the camera is in ?

Reply:
can we have the trigger details implemented in another PR @YangLiujames ? so we can move forward with merging earlier

Reply:
let's have the trigger related functionality in a separate PR later @YangLiujames , with a well tested daq card we can test those functionalities better.